app/forms/translatable_base_form.rb

```diff
@@ -0,0 +1,30 @@
+class TranslatableBaseForm < SimpleDelegator
+
+  extend ActiveModel::Translation
+
+  def self.for(instance)
+    instance_class = instance.class
+    # Dynamically create a subclass where the model_name is based off the original
+    # instance's type so that `human_attribute_name` can work off the same translations
+    # as the original object.
+    Class.new(self) do
```

Comment:
Please let me know if I've gone too meta here. It's definitely a little tricksy, but makes for an object that is virtually indistinguishable from the object itself in regards to the view and its translations.

I went kind of meta because we need a class-level `model_name` method in order for `human_attribute_name` to work, which is what simple_form uses under the hood for its labels. We usually don't need something quite so tricky because our form objects are a one-to-one mapping from a form to a single class. This one should support any of the subclasses of product.

Reply:
> Please let me know if I've gone too meta here.
😆 

I definitely am not a fan of this level of trickery and working around what SimpleForm and I18n expect in order to be able to trick them, as it’s usually a smell that there’s probably a better way to do this. Is there another way to organize this code that doesn’t require all this hiding and injecting ourselves in the method chain?

Reply:
As an aside, since this is the first time we’re doing this work, can we maybe not introduce a class hierarchy from the get-go? The inheritance and base class here seem a bit YAGNI to me.

What if it’s just `
class EditProductForm` and all the methods are there implemented only for the scope we currently have? Does that simplify things? ¢2